src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -85,9 +85,41 @@ export function groupTrialMatchesByAgeNumerical(armGroup: ITrialMatch[]): IClini
         clinicalGroupMatch.notMatches = positiveAndNegativeMatches.notMatches;
         return clinicalGroupMatch;
     });
+    if (matches.length > 1) {
+        return removeDuplicationByAge(matches);
+    }
     return matches;
 }
 
+export function removeDuplicationByAge(clinicalGroupMatch: IClinicalGroupMatch[]): IClinicalGroupMatch[] {
```

Comment:
this function could use a lot more commenting to explain what's going on.

Reply:
i think this code would be clearer and more performant if you first did groupBy using as a comparator the serialized positive and negative array sorted/concated together.  now you have the duplicates together and you can iterate through and do the merge on the age.  then just flatten it.